builder/src/lib.rs

```diff
@@ -1,10 +1,247 @@
 extern crate proc_macro;
+use std::collections::{HashMap, HashSet};
 
+use quote::{quote};
 use proc_macro::TokenStream;
+use proc_macro2::{TokenStream as TokenStream2};
+use syn::{DeriveInput, parse_macro_input, Data, Fields, Ident};
 
-#[proc_macro_derive(Builder)]
+#[proc_macro_derive(Builder, attributes(builder))]
 pub fn derive(input: TokenStream) -> TokenStream {
-    let _ = input;
+    let input = parse_macro_input!(input as DeriveInput);
+    let name = &input.ident;
+    let buildername = Ident::new(&format!("{}Builder", name), input.ident.span().clone());
 
-    unimplemented!()
+    let (fields, fields_init, setters, build) = match input.data {
```

Comment:
I think this would be more readable if we use the match-expression only for matching and not for doing work inside of it. Something like:

```rust
let fields = match input.data {
    ...
            Fields::Named(fields) => fields,
};

let builders = ...;
let fields = ...;
let fields_init = ...;
```